src/sentry/integrations/github/integration.py

```diff
@@ -426,11 +423,76 @@ def dispatch(self, request: Request, pipeline: Pipeline) -> HttpResponse:
                         "success": False,
                         "data": {"error": _("Github installed on another Sentry organization.")},
                     },
-                    "document_origin": document_origin,
+                    "document_origin": get_document_origin(self.active_organization),
                 },
                 request=request,
             )
 
         # OrganizationIntegration does not exist, but Integration does exist.
         pipeline.bind_state("installation_id", request.GET["installation_id"])
         return pipeline.next_step()
+
+
+class GitHubUserValidation(PipelineView):
```

Comment:
Can we write some tests for this validator?

for each step in the dispatch where we throw an error, we could mock the state of the client and assert that the error is thrown?